### PR TITLE
Remove modelcards dependency

### PR DESCRIPTION
**Pull request by Wauplin**
While working on https://github.com/huggingface/diffusers/pull/2047, I realized that `diffusers` depends on `modelcards` package [which is deprecated and not maintained anymore](https://github.com/nateraw/modelcards). This PR switches to the `huggingface_hub` implementation -a copy of `modelcards` with a few fixes-. It has been integrated to `hfh` in version `0.10` which is already a requirement for `diffusers`.

To make it work, one must have `Jinja2` installed [which was already a requirement](https://github.com/nateraw/modelcards/blob/main/requirements.txt#L2). I've updated the `create_model_card` helper and wrote a test for it. Actually I'm not sure it was working at all as the template path seemed to be wrong.

(this PR also removed `is_modelcards_available` boolean method. Idk the deprecation policy in `diffusers` but I can put it back if needed. I have added a `is_jinja_available` (from `huggingface_hub.utils`) which is more appropriate now.

Note: this is not related to the `0.12` release of `huggingface_hub`. It can be merged before or after it doesn't matter (or never)

### CODE REVIEW

File: src/diffusers/utils/hub_utils.py
Comment (patrickvonplaten) [inline src/diffusers/utils/hub_utils.py:86]: nice!

File: src/diffusers/utils/import_utils.py
Comment (patrickvonplaten) [inline src/diffusers/utils/import_utils.py:24]: Just to be sure this is supported since huggingface_hub `0.10` ? That's currently our minimum required version for the Hub.

Comment (patrickvonplaten) [reply src/diffusers/utils/import_utils.py:24]: Ah yeah you answered that above

Comment (Wauplin) [reply src/diffusers/utils/import_utils.py:24]: [Yes it is !](https://github.com/huggingface/huggingface_hub/blob/v0.10.0/src/huggingface_hub/utils/__init__.py#L56)

**Review by patrickvonplaten — APPROVED** [review]
Very clean! Good to merge for me. Let's wait until the tests are green and then we can merge :-)

**Comment by Wauplin** [comment]
That was a quick review! :smile: Thanks

**Pull request merged**
